odl/tomo/geometry/spect.py

```diff
@@ -70,6 +69,10 @@ def __init__(self, apart, dpart, det_rad, axis=[0, 0, 1],
         init_pos_norm = np.linalg.norm(orig_to_det_init)
         if init_pos_norm > 1e-10:
             orig_to_det_init *= self.det_radius / init_pos_norm
+        else:
+            raise ValueError('initial detector position {} is close to '
+                             'zero. This is only allowed for explicit '
+                             'det_init_axes.'.format(orig_to_det_init))
```

Comment:
The `det_init_axes` part slipped through, please remove. The first sentence can simply be `'initial detector position {} is too close to zero.'`